tests/test_legacy_api.py

```diff
@@ -1367,6 +1367,49 @@
         "events",
         id="wrapped_tag_functions",
     ),
+    pytest.param(
+        {
+            "selected_columns": [
+                [
+                    "if",
+                    [
+                        ["in", ["tags[sentry:release]", "tuple", ["'foo'"]]],
+                        "tags[sentry:release]",
+                        "'other'",
+                    ],
+                    "release",
+                ]
+            ],
```

Comment:
Could you provide an example of how this was parsed before ? I am not sure about about what the specific issue was. I remember ambiguities in the parsing, but cannot remember the details.

Reply:
The original parser would fail because it would recurse through the first term until it got to `"in", ["tags[sentry:release]", "tuple", ["'foo'"]]]`, where it would call `parse_exp` on each of the terms individually. This would throw an exception because `["'foo'"]` is a list, which it would try to interpret as a function and fail.

In Snuba, the parser iterates through the arguments by one but also by two, and evaluate whether those two arguments are in turn a function.